### PR TITLE
add collectiv section

### PR DESCRIPTION
**Pull request by tiwarishankar**
Here are the changes I've made:

1. Added one hook for checking mobile devices
2. Added two component for collectiv branding.

Here is the demo of the feature.

https://github.com/lucavallin/first-issue/assets/71537738/1434441b-ad14-4b2e-af4d-389b9e9f4bb7

### CODE REVIEW

**Review by SubhamPramanik — APPROVED** [review]

File: components/Collectiv/CollectivSection.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export const CollectivSection = () => {
+  return (
+    <div className="pt-6">
+      <div className="flex cursor-pointer flex-row items-center text-secondary">GET HELP</div>
```
Comment (lucavallin) [inline components/Collectiv/CollectivSection.tsx:4]:
```suggestion
      <div className="flex cursor-pointer flex-row items-center text-secondary uppercase">Get Help</div>
```

File: components/Collectiv/CollectivSection.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            Collectiv
+            <img
+              src="/sponsors/linkout.png"
+              alt="link"
+              className="mb-1 ml-1 inline-block h-[12px] w-[12px]"
+            />
+          </span>{" "}
+        </a>
+        to solve Hacktoberfest issues and{" "}
+        <span className="mb-2 text-sm font-bold  tracking-wider text-[#F8F8F8]">win a T-shirt</span>{" "}
```
Comment (lucavallin) [inline components/Collectiv/CollectivSection.tsx:22]:
```suggestion
        <span className="mb-2 text-sm font-bold tracking-wider text-[#F8F8F8]">win a T-shirt</span>{" "}
```

File: components/Collectiv/CollectivSection.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      <div className=" py-2">
+        Use{" "}
+        <a
+          href="https://chat.collectivai.com/hacktober-fest"
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          <span className="font-bold text-primary transition-all hover:underline">
+            Collectiv
+            <img
```
Comment (lucavallin) [inline components/Collectiv/CollectivSection.tsx:14]: You can use `FortAwesome` icons instead

File: components/Collectiv/RepositoryCollectivButton.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      className={`flex items-center text-sm  font-semibold ${
+        isMedia720px
+          ? "m-2 justify-center gap-2 rounded-[4px]  border  border-primary px-[8px] py-[6px]"
+          : "mx-2 ml-2 gap-1 rounded-full  border-transparent bg-primary px-[14px] text-light-200"
+      }`}
+      target="_blank"
+      rel="noopener noreferrer"
+    >
+      Get help from Collectiv
+      <img
```
Comment (lucavallin) [inline components/Collectiv/RepositoryCollectivButton.tsx:27]: You can use `FortAwesome` icons instead

File: components/Repository/IssueList.tsx [hunk 5]
```diff
@@ -1,16 +1,31 @@
...
 
 type IssueListProps = {
   issues: Issue[];
+  repositoryName: Repository["name"];
+  repositoryOwner: Repository["owner"];
 };
 
-export const IssuesList = ({ issues }: IssueListProps) => {
+export const IssuesList = ({ issues, repositoryName, repositoryOwner }: IssueListProps) => {
+  const isMedia720px = useMediaQuery("(max-width : 720px)");
```
Comment (lucavallin) [inline components/Repository/IssueList.tsx:14]: See comment about tailwind's breakpoint prefixes

File: public/sponsors/collectivIcon.png
Comment (lucavallin) [inline public/sponsors/collectivIcon.png:1]: `collectiv.png`

File: components/SponsorsBar.tsx [hunk 6]
```diff
@@ -9,6 +9,9 @@ export const SponsorsBar = () => (
       <Link href="https://lucavall.in" className="grayscale hover:grayscale-0">
         <Image src="/sponsors/lucavallin.png" alt="lucavallin" width={25} height={25} />
       </Link>
+      <Link href="https://chat.collectivai.com/" className="hover:scale-110">
```
Comment (lucavallin) [inline components/SponsorsBar.tsx:12]: let's make sure all sponsor icons follow the same style:

- gray
- color &bigger when hover (but it should not move the page)

File: components/Sidebar.tsx [hunk 7]
```diff
@@ -34,6 +35,7 @@ export const Sidebar = () => {
       <LinkButton href="https://github.com/lucavallin/first-issue#adding-a-new-project">
         Add your project
       </LinkButton>
+      <CollectivSection />
```
Comment (lucavallin) [inline components/Sidebar.tsx:38]: I am not a fan of this one, it really pushes the browse tabs way down ;) alternative locations?

File: components/Collectiv/RepositoryCollectivButton.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+      href={`https://chat.collectivai.com/contribute?repo-name=${ownerAndRepoName}`}
+      className={`flex items-center text-sm  font-semibold ${
+        isMedia720px
+          ? "m-2 justify-center gap-2 rounded-[4px]  border  border-primary px-[8px] py-[6px]"
+          : "mx-2 ml-2 gap-1 rounded-full  border-transparent bg-primary px-[14px] text-light-200"
+      }`}
+      target="_blank"
+      rel="noopener noreferrer"
+    >
+      Get help from Collectiv
```
Comment (lucavallin) [inline components/Collectiv/RepositoryCollectivButton.tsx:26]: This should be shorter, it takes a lot of space

File: components/Repository/RepositoryItem.tsx [hunk 9]
```diff
@@ -52,7 +52,13 @@ export const RepositoryItem = ({ repository }: RepositoryItemProps) => {
           repositoryStars={repository.stars_display}
         />
       </div>
-      {isIssueOpen && <IssuesList issues={repository.issues} />}
+      {isIssueOpen && (
+        <IssuesList
+          issues={repository.issues}
+          repositoryName={repository.name}
+          repositoryOwner={repository.owner}
```
Comment (lucavallin) [inline components/Repository/RepositoryItem.tsx:59]: These two could be just one property?

File: components/Repository/RepositoryItemTopBar.tsx [hunk 10]
```diff
@@ -30,6 +34,12 @@ export const RepositoryItemTopBar = ({
       <div className="flex flex-1 items-center justify-end">
         <div className={`h-2 w-2 rounded-full ${repositoryHasNewIssues ? "bg-primary" : ""}`}></div>
       </div>
+      {isIssueOpen && !isMedia720px && (
+        <RepositoryCollectivButton
```
Comment (lucavallin) [inline components/Repository/RepositoryItemTopBar.tsx:38]: Will it work with GitLab repos?

File: components/Sidebar.tsx (hunk 7 above)
Comment (Nischalj10) [reply components/Sidebar.tsx:38]: Hey @lucavallin, we tried a few design iterations but other places were not looking good as well. Suggestion - we can keep this for October and till then really try a few more iterations on design for an alternative place.

If this this cannot go for October as well, then we can go ahead and comment it out.

File: components/Repository/RepositoryItemTopBar.tsx (hunk 10 above)
Comment (Nischalj10) [reply components/Repository/RepositoryItemTopBar.tsx]: Will introduce a flow on our end for handling gitlab repos! Thanks.
@tiwarishankar please handle gitlab redirects it on Collectiv.

Comment (tiwarishankar) [reply components/Repository/RepositoryItemTopBar.tsx]: yes added here for gitlab repo!!

File: components/Sidebar.tsx (hunk 7 above)
Comment (lucavallin) [reply components/Sidebar.tsx:38]: Sounds good!

**Comment by lucavallin** [comment]
image

Font is different here.

**Review by lucavallin — APPROVED** [review]
🥳 Awesome, thanks!

**Pull request merged**